src/main/java/org/opentripplanner/routing/core/RoutingRequest.java

```diff
@@ -652,6 +658,13 @@ public void setBannedRoutes(String s) {
             bannedRoutes = RouteMatcher.emptyMatcher();
     }
 
+    public void setWhiteListedRoutes(String s) {
+        if (s != null && !s.equals(""))
```

Comment:
The canonical way of performing the second check is `String.isEmpty()`. If we are going to check for equality with the empty string, note that by reversing the comparison we can elide the null check: `!("".equals(string))`